### PR TITLE
chore(cypress): TS Migration: projects.ts, show-cert-from-superblock.ts and challenges.ts

### LINKED ISSUE #48753: TypeScript Migrate Cypress Tests
## Guidelines

Take a look at the [Codebase Best Practices](https://contribute.freecodecamp.org/#/codebase-best-practices) - **kebab-case file names**

### Do Not Ignore Linting, Unless **Absolutely** Necessary

We hope to leverage the type-safety that comes with TypeScript. So, do not ignore the compiler's warnings/errors. If you are stuck, feel free to reach out for help.

### Keeping Git History

Sometimes changing the file from `<filename>.js` -> `<filename>.ts?` causes the original file to be deleted, and a new one created, and other times the filename just changes - in terms of Git. Ideally, we want the file history to be preserved.

The best bet at achieving this is to:
1) Rename the file
2) Commit with the flag `--no-verify` to prevent husky from complaining about the errors
3) Refactor to TS for migration, in a separate commit.

**Note:** _Editors like VSCode are still likely to show you the file has been deleted and a new one created. If you use the CLI to `git add .`, then VSCode will show the file as renamed in stage_

## Files to be migrated

- [x] ./cypress/e2e/default/learn/challenges/multifile.js
- [x] ./cypress/e2e/default/learn/challenges/sass.js
- [x] ./cypress/e2e/default/learn/challenges/project-preview.js
- [x] ./cypress/e2e/default/learn/challenges/multifileCertProject.js
- [x] ./cypress/e2e/default/learn/challenges/failed-updates.js
- [ ] /cypress/e2e/default/learn/challenges/projects.js
- [x] ./cypress/e2e/default/learn/challenges/output.js
- [x] ./cypress/e2e/default/learn/coding-interview-prep/intro-page.js
- [x] ./cypress/e2e/default/learn/common-components/navbar.js
- [x] ./cypress/e2e/default/learn/common-components/searchBar.js
- [x] ./cypress/e2e/default/learn/common-components/editor.js
- [x] ./cypress/e2e/default/learn/common-components/footer.js
- [x] ./cypress/e2e/default/learn/common-components/helpButton.js
- [x] ./cypress/e2e/default/learn/donate/donation-block-completion-modal.js
- [x] ./cypress/e2e/default/learn/donate/donate-page-default.js
- [x] ./cypress/e2e/default/learn/donate/donate-page-donor.js
- [x] ./cypress/e2e/default/learn/header/universal-navigation.js
- [ ] ./cypress/e2e/default/learn/redirects/heading-challenge.js
- [x] ./cypress/e2e/default/learn/redirects/es6-to-basic-javascript.js
- [x] ./cypress/e2e/default/learn/redirects/breadcrumbs.js
- [x] ./cypress/e2e/default/learn/responsive-web-design/intro-page.js
- [x] ./cypress/e2e/default/learn/responsive-web-design/basic-css/index.js
- [ ] ./cypress/e2e/default/learn/responsive-web-design/show-cert-from-superblock.js
- [x] ./cypress/e2e/default/learn/challenge-hot-keys.js
- [x] ./cypress/e2e/default/learn/index.js
- [ ] ./cypress/e2e/default/legacy/redirects/challenges.js
- [ ] ./cypress/e2e/default/legacy/redirects/adding-development.js
- [ ] ./cypress/e2e/default/settings/email-change.js
- [x] ./cypress/e2e/default/settings/user-token.js
- [x] ./cypress/e2e/default/settings/username-change.js
- [x] ./cypress/e2e/default/settings/certifications.js
- [x] ./cypress/e2e/default/settings/image-picture-check.js
- [x] ./cypress/e2e/default/settings/settings.js
- [x] ./cypress/e2e/default/user/report-user.js
- [x] ./cypress/e2e/default/user/privacy-terms.js
- [x] ./cypress/e2e/default/user/certifications.js
- [x] ./cypress/e2e/default/landing.js
- [ ] ./cypress/e2e/default/ShowCertification.js
- [ ] ./cypress/e2e/default/tags.js
- [x] ./cypress/e2e/default/top-contributor.js
- [ ] ./cypress/e2e/mobile-learn/test-challenges.js
- [ ] ./cypress/e2e/third-party/donate-page.js
- [ ] ./cypress/plugins/index.js

### PR DESCRIPTION
**Pull request by pmelin**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #48753

On `projects.ts` I looked around the code to see if I could find an existing `Curriculum` type to use but couldn't find any, so I created the types for the test.

On `show-cert-from-superblock.ts` was a simple fix of wrapping an unknown type into a String.

On `challenges.ts` there was a conflict using the `location` variable name (cypress/e2e/default/learn/responsive-web-design/basic-css/index.ts) so I renamed to `testLocations`. 

Please let me know if you wish any adjustments. :+1:

### CODE REVIEW

File: cypress/e2e/default/learn/challenges/projects.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
+import { SuperBlocks } from '../../../../../config/certification-settings';
+
+interface Meta {
+  challengeOrder: string[][];
+}
+
+interface Solution {
+  contents: unknown;
+  fileKey: string;
+}
```
Comment (ojeytonwilliams) [inline cypress/e2e/default/learn/challenges/projects.ts:10]:
```suggestion
interface File {
  contents: unknown;
  fileKey: string;
}

type Solution = File[];
```
A solution is an array of these types, so it's a little confusing if this is called "Solution".

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Nice work, @pmelin. I just found one place where I think the types could be clearer:

**Comment by ojeytonwilliams** [comment]
> On projects.ts I looked around the code to see if I could find an existing Curriculum type to use but couldn't find any, so I created the types for the test.

Thanks for taking the time to do that. We do need to provide types for the curriculum at some point, but this is totally fine for now.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks for making those updates.

This all LGTM :+1:

**Review by Sboonny — APPROVED** [review]
#### :sparkles: :tada: **AWESOME!** :tada: :sparkles:

 Hi @pmelin, thanks for this pull request and for contributing to the code-base for the first time. We are looking forward to more contributions from you in the future.

 Cheers & happy contributing!

**Pull request merged**